File: 2130-maximum-twin-sum-of-a-linked-list/2130-maximum-twin-sum-of-a-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def pairSum(self, head: Optional[ListNode]) -> int:
        if head is None:
            return None
        stack=[]
        fast=slow=head
        while(fast!=None and fast.next!=None):
            fast=fast.next.next
            slow=slow.next
        mid=slow
        while(slow!=None):
            stack.append(slow)
            slow=slow.next
        curr=head
        ans=0
        while(curr!=mid):
            ans=max(ans,curr.val+stack.pop().val)
            curr=curr.next
        return ans
            
        
        

File: 2130-maximum-twin-sum-of-a-linked-list/2130-maximum-twin-sum-of-a-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    int pairSum(ListNode* head) {
        vector<int>Arr;
        ListNode*half=head;
        while(head!=nullptr and head->next!=nullptr)
        {
            Arr.push_back(half->val);
            head=head->next->next;
            half=half->next;
        }
        int ans=0;
        int i=Arr.size()-1;
        while(half!=nullptr)
        {
            ans=max(ans,Arr[i]+half->val);
            i--;
            half=half->next;
        }
        return ans;
    }
};